perf/terraform/configs/local/terraform.tf

```diff
@@ -0,0 +1,121 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.67.0"
+    }
+  }
+}
+
+locals {
+  tags = {
+    Project = "perf"
+  }
+}
+
+provider "aws" {
+  alias  = "us-west-2"
+  region = "us-west-2"
+  default_tags {
+    tags = local.tags
+  }
+}
+
+provider "aws" {
+  alias  = "us-east-1"
+  region = "us-east-1"
+  default_tags {
+    tags = local.tags
+  }
+}
+
+
+variable "ci_enabled" {
+  type        = bool
+  description = "Whether or not to create resources required to automate the setup in CI (e.g. IAM user, cleanup Lambda)"
+  default     = false
+}
+
+variable "long_lived_enabled" {
+  type        = bool
+  description = "Whether or not to create long lived resources (in CI, used across runs; e.g. VPCs)"
+  default     = true
+}
+
+variable "short_lived_enabled" {
+  type        = bool
+  description = "Whether or not to create short lived resources (in CI, specific to each run; e.g. EC2 instances)"
+  default     = true
+}
```

Comment:
Playing with this, I am still a bit confused.

If I understand correctly the `.github/workflows/perf` workflow will run the `perf/terraform/local` config, see above. That config has `long_lived_enabled` and `short_lived_enabled` set to `true`. The latter makes sense. Why does it set the former? Shouldn't it use the long-lived resources created outside of the workflow?

Looking at it from a different perspective, don't we really need 3 configs?

1. `configs/local` for local development with `ci_enabled` `false`, `long_lived_enabled` `true`, `short_lived_enabled` `true` and no S3 backend.
2. `configs/ci_long_lived` run once outside of the GitHub workflow with `ci_enabled` `true`, `long_lived_enabled` `true`, `short_lived_enabled` `false` and S3 backend.
3. `configs/ci_short_lived` run once outside of the GitHub workflow with `ci_enabled` `false`, `long_lived_enabled` `false`, `short_lived_enabled` `true` and no S3 backend.

Reply:
That's a very valid question because it's not immediately obvious why we have different configs in the first place. And the answer lies within terraform internals which we definitely shouldn't require users to know.

So the terraform specific thing I'm referring to here is that terraform backend (i.e. config of where the state is stored) cannot be defined dynamically. That's the reason why we need 2 separate configs - one where state is stored locally, and the other where the state is stored remotely (in this case, on S3). On the other hand, what modules are created during terraform apply can be configured dynamically through terraform variables. That's why we only need 2 configs.

So how are we running tf and where?
1. We use `config/local` during local development. `config/local` defaults are set so that only `long_lived` and `short_lived` resources are created. Because of that, the user doesn't have to specify any additional input when doing terraform apply or destroy. The state is stored locally.
2. We use `config/remote` when we want to do the initial set up of the resources required for the CI. By default, this config creates only `ci` and `long_lived` resources. Again, no additional input is required when we use this config. The state is stored remotely, in a S3 bucket.
3. We use `config/local` when we want to start instances for CI runs. We use `config/local` because we want the state to be stored locally but `config/local`, by default, creates `long_lived` and `short_lived` resources. This is too much so we have to override the defaults. This is fine because we don't put the burden of overriding defaults on users. Instead, we just configure it once in the workflow YAML via `TF_VAR_*` env vars, see https://github.com/libp2p/test-plans/blob/020d5097fecb4c6e2c271d8748bb5841fe0d9e1a/.github/workflows/perf.yml#L35-L37

Because it's a bit tricky, I included a README next to the configs: https://github.com/libp2p/test-plans/blob/020d5097fecb4c6e2c271d8748bb5841fe0d9e1a/perf/terraform/configs/README.md